atom/browser/api/atom_api_power_save_blocker.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2015 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#ifndef ATOM_BROWSER_API_ATOM_API_POWER_SAVE_BLOCKER_H_
+#define ATOM_BROWSER_API_ATOM_API_POWER_SAVE_BLOCKER_H_
+
+#include "base/memory/scoped_ptr.h"
+#include "content/public/browser/power_save_blocker.h"
+#include "native_mate/handle.h"
+#include "native_mate/wrappable.h"
+
+namespace mate {
+class Dictionary;
+}
+
+namespace atom {
+
+namespace api {
+
+class PowerSaveBlocker : public mate::Wrappable {
+ public:
+  static mate::Handle<PowerSaveBlocker> Create(v8::Isolate* isolate);
+
+ protected:
+  PowerSaveBlocker();
+  virtual ~PowerSaveBlocker();
+
+  // mate::Wrappable implementations:
+  mate::ObjectTemplateBuilder GetObjectTemplateBuilder(
+      v8::Isolate* isolate) override;
+
+ private:
+  void Start(content::PowerSaveBlocker::PowerSaveBlockerType type);
```

Comment:
This probably should return some kind of token you then pass to `Stop`, or at least does a refcount thing so that two `Start`s require two `Stop`s to stop. Otherwise it's very hard to use this in a library without worrying if you're stomping somebody else's Power Save Block - i.e:

1. App calls Start for a long process
1. Some library also calls Start, for something short
1. That library calls Stop
1. Library accidentally stomps the App's Start call and has no way to _not_ do that